tiledb/sm/cpp_api/array.h

```diff
@@ -598,6 +628,22 @@ class Array {
     return timestamp;
   }
 
+  /** Sets the array config. */
+  void set_config(const Config& config) const {
+    auto& ctx = ctx_.get();
+    ctx.handle_error(tiledb_array_set_config(
+        ctx.ptr().get(), array_.get(), config.ptr().get()));
+  }
+
+  /** Retrieves the array config. */
+  Config get_config() const {
```

Comment:
Rename to `config()`? This is what we usually do as a convention for getters that return an object (as opposed to getting it from a pointer passed as an argument).